native/src/routes/search/components/__tests__/SearchModal.spec.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+
+import React from 'react'
+import { fireEvent, render } from '@testing-library/react-native'
+import SearchModal from '../SearchModal'
+import lightTheme from '../../../../modules/theme/constants'
+import type { CategoriesRouteInformationType } from 'api-client'
+import { CATEGORIES_ROUTE, SEARCH_FINISHED_SIGNAL_NAME } from 'api-client'
+import CategoriesMapModelBuilder from 'api-client/src/testing/CategoriesMapModelBuilder'
+import sendTrackingSignal from '../../../../modules/endpoint/sendTrackingSignal'
+import { urlFromRouteInformation } from '../../../../modules/navigation/url'
+
+// $FlowFixMe
+import { ThemeProvider } from 'styled-components/native'
+
+jest.mock('rn-fetch-blob')
+jest.mock('../../../../modules/endpoint/sendTrackingSignal')
+
+describe('SearchModal', () => {
+  beforeEach(() => {})
+
+  const dummy = jest.fn()
+  const t = key => key
+  const categoriesMapModel = new CategoriesMapModelBuilder('augsburg', 'de').build()
+  const language = 'de'
+  const cityCode = 'augsburg'
+  const props = {
+    categories: categoriesMapModel,
+    navigateTo: dummy,
+    language: language,
+    cityCode: cityCode,
+    closeModal: dummy,
+    navigateToLink: dummy,
+    t: t,
+    theme: lightTheme,
+    sendFeedback: dummy
+  }
+
+  it('should send tracking signal when closing search site', async () => {
+    const { getByPlaceholderText, getAllByRole } = render(
+      <ThemeProvider theme={lightTheme}>
+        <SearchModal {...props} />
+      </ThemeProvider>
+    )
+    const button = getAllByRole('button')[0]
+    const searchBar = getByPlaceholderText('searchPlaceholder')
+    await fireEvent.changeText(searchBar, 'Category')
+    await fireEvent.press(button)
+    expect(sendTrackingSignal).toHaveBeenCalledTimes(1)
+
```

Comment:
Added this test for closing the search, but it fails an i have no idea why. It says, that sendTrackingSignal is not called, but when debugging it the code is executed, but somehow the test does not register it.

Reply:
And even though the test below doesn't fail, you basically create the same situation with `onItemPress`. The function execution order is pretty unforseeable for asynchronous functions, when they are not awaited. That's why I'd propose to also apply the solution to the other test.

Reply:
Thank you very much. Now it works :)